src/mds/Server.cc

```diff
@@ -785,6 +785,7 @@ void Server::handle_client_session(const cref_t<MClientSession> &m)
 	mds->locker->resume_stale_caps(session);
 	mds->sessionmap.touch_session(session);
       }
+      trim_completed_request_list(m->oldest_client_tid, session);
```

Comment:
what about the `FLUSH_ACK` that is usually sent back for a `FLUSH` req ?
isn't that necessary in this situation ?
maybe we should send back the `oldest_tid` in the renew caps ack msg and act on the oldest tid on the client side as if a `FLUSH_ACK` was was received

Reply:
The `flush ack` sounds not reliable, we couldn't guarantee that this will always work. But for the session's renew caps msg we can be sure that this will be sent per tick. This will also be used to keep the client alive.

Reply:
Also please note the `WRITE` requests do not mean the real `write` contents to Rados, the `WRITE` means the request ops have the bit of `#define CEPH_MDS_OP_WRITE        0x001000`. Such as `create/setattr,etc`.

So if all the `WRITE` requests are sync create ops, it seems there has no cap flush or flush ack between MDS and clients.